bliss/simulator/prior.py

```diff
@@ -139,11 +207,35 @@ def _sample_n_galaxies_and_stars(self, n_sources):
 
         return galaxy_bools, star_bools
 
-    def _sample_star_fluxes(self):
-        latent_dims = (self.batch_size, self.n_tiles_h, self.n_tiles_w, self.max_sources, 1)
-        return self._draw_truncated_pareto(
+    def _sample_star_fluxes(self, star_ratios):
+        latent_dims = (self.batch_size, self.n_tiles_h, self.n_tiles_w, self.max_sources)
+        fluxes = torch.zeros(
+            self.batch_size, self.n_tiles_h, self.n_tiles_w, self.max_sources, self.n_bands
+        )
+        # sample value for first band from pareto for each light source
+        fluxes[:, :, :, :, 0] = self._draw_truncated_pareto(
             self.star_flux_alpha, self.star_flux_min, self.star_flux_max, latent_dims
         )
+        # Compute remaining band fluxes via ratio from SDSS catalog and sampled values
+        if self.n_bands > 1:
+            fluxes[:, :, :, :, 1:] = self.multiflux(star_ratios, fluxes[:, :, :, :, 0])
+        return fluxes
+
+    def multiflux(self, rcf_ratios, flux_base) -> Tensor:
+        """Generate flux values for remaining bands based on draw."""
+        flux_rest = torch.zeros(
+            self.batch_size, self.n_tiles_h, self.n_tiles_w, self.max_sources, self.n_bands - 1
+        )
```

Comment:
Why is so much of this code not run by our existing tests? Could we change the code so that this code does get exercised by our existing tests? i.e., is possible, I'd like to treat single-band generation as special case of multi-band generation that doesn't require different control flow.

Reply:
It's harder for bugs to hide is there aren't two ways to do it: one for single-band and one for multi-band. I'd like the same multi-band generation code to work for any number of bands: such as 5, 3, or even 1.

Reply:
I think in the config located on the remote repo the number of bands is specified as just 1 which is probably why CodeCov didn't check any of the lines where `n_bands` was greater than 1. When testing locally prior to committing, the tests hit the unchecked code when I was debugging, so I might change the base and testing configs to specify multiple bands. I definitely agree that the focus should be multi-band generation and that single-band is a special case. 

I thought to add the conditional statements because we don't have a reason to do any sampling from the SDSS catalogs for ratios if we are only using a single band. I'm not sure how we could exactly combine the multi-band and single-band cases for that reason.

Reply:
Even in the single band case, if that single band isn't the r band, we'll still need to use this sampling code. I'm not sure if this, but perhaps we should sample all five bands each time, and then only use the fluxes for subset of the bands that are specified in the config file.

Reply:
Oh. So is the Pareto distribution we sample from only parameterized to reflect the r-band flux value?

Reply:
Yes, I think so

Reply:
@sawanp813 Could we just always load the ratios for all five bands in the constructor and return all five bands in `multiflux`, and then just slice by the bands we want in `sample_prior`? So `sample_star_fluxes` and `sample_galaxy_fluxes` would always return a tensor with last dimension 5 for all 5 bands, and then in sample_prior we can do `self._sample_galaxy_prior(select_gal_rcfs)[:, :, :, :, self.bands]`
https://github.com/prob-ml/bliss/blob/95e3087cf69e40da628e33060245e09b851d1d5e/bliss/simulator/prior.py#L167-L169

It would be a bit inefficient for the one-band case, but it's only during generation and probably not that likely to be used.

Reply:
Right, that makes sense if the Pareto distribution only reflects the r-band. We can definitely make that change.

Reply:
It would be inefficient for the one-band case but it doesn't seem like we'll be using single-band inference going forward anyways. I feel like that would be the best way to handle it.